modules/taboolaBidAdapter.js

```diff
@@ -0,0 +1,260 @@
+
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import {config} from '../src/config.js';
+import {getWindowSelf, getWindowTop} from '../src/utils.js'
+import {getStorageManager} from '../src/storageManager.js';
+
+const BIDDER_CODE = 'taboola';
+const GVLID = 42;
+const CURRENCY = 'USD';
+export const END_POINT_URL = 'http://hb.bidder.taboola.com/TaboolaHBOpenRTBRequestHandlerServlet';
+const USER_ID = 'user-id';
+const STORAGE_KEY = `taboola global:${USER_ID}`;
+const COOKIE_KEY = 'trc_cookie_storage';
+
+/**
+ *  extract User Id by that order:
+ * 1. local storage
+ * 2. first party cookie
+ * 3. rendered trc
+ * 4. new user set it to 0
+ */
+export const userData = {
+  storageManager: getStorageManager({gvlid: GVLID, bidderCode: BIDDER_CODE}),
+  getUserId: () => {
+    const {getFromLocalStorage, getFromCookie, getFromTRC} = userData;
+
+    try {
+      return getFromLocalStorage() || getFromCookie() || getFromTRC();
+    } catch (ex) {
+      return 0;
+    }
+  },
+  getFromCookie() {
+    const {cookiesAreEnabled, getCookie} = userData.storageManager;
+    if (cookiesAreEnabled()) {
+      const cookieData = getCookie(COOKIE_KEY);
+      const userId = userData.getCookieDataByKey(cookieData, USER_ID);
+      if (userId) {
+        return userId;
+      }
+    }
+  },
+  getCookieDataByKey(cookieData, key) {
+    const [, value = ''] = cookieData.split(`${key}=`)
+    return value;
+  },
+  getFromLocalStorage() {
+    const {hasLocalStorage, localStorageIsEnabled, getDataFromLocalStorage} = userData.storageManager;
+
+    if (hasLocalStorage() && localStorageIsEnabled()) {
+      return getDataFromLocalStorage(STORAGE_KEY);
+    }
+  },
+  getFromTRC() {
+    return window.TRC ? window.TRC.user_id : 0;
+  }
+}
+
+export const internal = {
+  getPageUrl: (refererInfo = {}) => {
+    if (refererInfo.canonicalUrl) {
+      return refererInfo.canonicalUrl;
+    }
+
+    if (config.getConfig('pageUrl')) {
+      return config.getConfig('pageUrl');
+    }
+
+    try {
+      return getWindowTop().location.href;
+    } catch (e) {
+      return getWindowSelf().location.href;
+    }
+  },
+  getReferrer: (refererInfo = {}) => {
+    if (refererInfo.referer) {
+      return refererInfo.referer;
+    }
+
+    try {
+      return getWindowTop().document.referrer;
+    } catch (e) {
+      return getWindowSelf().document.referrer;
+    }
+  }
+}
+
+export const spec = {
+  supportedMediaTypes: [BANNER],
+  gvlid: GVLID,
+  code: BIDDER_CODE,
+  isBidRequestValid: (bidRequest) => {
+    return !!(bidRequest.sizes &&
+              bidRequest.params &&
+              bidRequest.params.publisherId &&
+              bidRequest.params.tagId);
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const [bidRequest] = validBidRequests;
+    const {refererInfo, gdprConsent = {}, uspConsent} = bidderRequest;
+    const {bcat = [], badv = [], publisherId} = bidRequest.params;
```

Comment:
Actually, since that release is right around the corner, we would prefer if you made the change to using the config ortb2 location for bcat/badv in the current pr.

Reply:
hey @ncolletti i updated the code so the badv and bcat will based on the config 
but it seems like the unit tests are super flaky 
(other adapters are failing and my change is so small that it can't be the reason ) so i cant get a green build also i didn't find a way to rerun the job but making a push to the brunch